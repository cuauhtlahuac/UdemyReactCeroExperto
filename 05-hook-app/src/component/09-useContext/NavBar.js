import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
	return (
		<nav>
			<ul className="nav nav-tabs">
				<li className="nav-item">
					<NavLink
           activeClassName="active"
           className="nav-link"
           to="/"
           exact
           >
						Home
					</NavLink>
				</li>
				<li className="nav-item">
					<NavLink
						activeClassName="active"
						className="nav-link"
						to="/login"
						exact
					>
						Login
					</NavLink>
				</li>
				<li className="nav-item">
					<NavLink
						activeClassName="active"
						className="nav-link"
						to="/about"
						exact
					>
						About
					</NavLink>
				</li>
			</ul>
		</nav>
	);
};

export default NavBar;
