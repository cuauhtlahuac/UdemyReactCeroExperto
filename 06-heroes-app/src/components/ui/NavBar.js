import React, { useContext } from 'react';
import { Link, NavLink } from 'react-router-dom';

import { AuthContext } from 'contexts/authContext';

const Navbar = () => {
	const { user } = useContext(AuthContext);

	return (
		<nav className="navbar navbar-expand-sm navbar-dark bg-dark p-3">
			<Link className="navbar-brand" to="/">
				Home
			</Link>

			<div className="navbar-collapse">
				<div className="navbar-nav">
					<NavLink
						activeClassName="active"
						className="nav-item nav-link"
						exact
						to="/marvel"
					>
						Marvel
					</NavLink>

					<NavLink
						activeClassName="active"
						className="nav-item nav-link"
						exact
						to="/dc"
					>
						DC
					</NavLink>

					<NavLink
						activeClassName="active"
						className="nav-item nav-link"
						exact
						to="/search"
					>
						Search
					</NavLink>
				</div>
			</div>

			<div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
				<ul className="navbar-nav ml-auto">
					<span className="nav-item nav-link text-info">{user.name}</span>
					<NavLink
						activeClassName="active"
						className="nav-item nav-link"
						exact
						to="/login"
					>
						Logout
					</NavLink>
				</ul>
			</div>
		</nav>
	);
};

export default Navbar;
