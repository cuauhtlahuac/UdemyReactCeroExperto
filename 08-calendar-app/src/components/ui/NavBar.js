const NavBar = ({ name }) => {
	return (
		<div className="navbar navbar-dark bg-dark mb-4">
			<span className="navbar-brand">{name}</span>
			<button className="btn btn-outline-danger">
				<i className="fas fa-sign-out-alt" /> <span>Salir</span>
			</button>
		</div>
	);
};

export default NavBar;
