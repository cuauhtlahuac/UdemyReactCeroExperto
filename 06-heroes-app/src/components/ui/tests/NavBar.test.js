import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import { AuthContext } from 'contexts/authContext';

import NavBar from '../NavBar';

describe('<NavBar />', () => {
	const contextValue = {
		dispatch: jest.fn(),
		user: { logged: false },
	};

	const component = mount(
		<AuthContext.Provider value={contextValue}>
			<MemoryRouter>
				<NavBar />
			</MemoryRouter>
		</AuthContext.Provider>,
	);

	test('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});

	test('should have a Link with home word', () => {
		const actual = component.find('a').first();

		expect(actual.text()).toMatch('Home');
	});
});
