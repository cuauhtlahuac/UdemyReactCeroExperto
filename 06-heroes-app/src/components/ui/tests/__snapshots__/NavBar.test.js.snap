// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark p-3"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      />
      <button
        className="nav-item nav-link btn btn-outline-primary"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
